doc/user-guides/oidcpolicy/oidc-keycloak.md

```diff
@@ -0,0 +1,238 @@
+# Securing Application with OIDCPolicy and Keycloak as IDP
+
+This tutorial demonstrates how to use the _OIDCPolicy_ extension to secure an application and authenticate a user implementing
+the [OpenID Connect Authorization Code Flow](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth) configuring [Keycloak](https://www.keycloak.org/) as the IDP.
+
+## Overview
+
+In this tutorial, you will:
+
+1. Set up a basic Gateway and HTTPRoute
+2. Deploy an example application
+3. Deploy Keycloak and use it as the IDP
+4. Apply an _OIDCPolicy_ to configure the AuthN/AuthZ for your application
+5. Test the AuthN/AuthZ functionality
+
+## Prerequisites
+
+- Kubernetes cluster with Kuadrant operator installed. See our [Getting Started](/latest/getting-started) guide for more information.
+- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) command line tool.
+
+## Notes
+
+- We'll be using [nip.io wildcard DNS](https://nip.io/) in order to simplify the guide without the need to set up DNS instances
+
+### Setup environment variables
+
+Set environment variables for convenience:
+
+```sh
+export KUADRANT_GATEWAY_NS=api-gateway      # Namespace for the Gateway
+export KUADRANT_GATEWAY_NAME=ingressgateway # Name for the Gateway
+export KUADRANT_DEVELOPER_NS=bakery         # Namespace for the app
+export KUADRANT_KEYCLOAK_NS=keycloak        # Namespace for keycloak
+```
+
+## Deploy Kuadrant and Gateway
+
+* Deploy the Kuadrant instance:
+
+```sh
+kubectl apply -f - <<EOF
+apiVersion: kuadrant.io/v1beta1
+kind: Kuadrant
+metadata:
+  name: kuadrant
+  namespace: kuadrant-system
+EOF
+```
+
+* Create the gateway namespace and deploy the Gateway:
+
+```sh
+kubectl create ns ${KUADRANT_GATEWAY_NS}
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: ${KUADRANT_GATEWAY_NAME}
+  namespace: ${KUADRANT_GATEWAY_NS}
+  labels:
+    kuadrant.io/gateway: "true"
+spec:
+  gatewayClassName: istio # Replace with `eg` for Envoy gateway
```

Comment:
```suggestion
  gatewayClassName: istio # Replace with "eg" for Envoy gateway
```
nit: since got the following when applying for me, but it applied fine anyhow

```
zsh: command not found: eg
```